File: Rinnerdos-cs50-problems-2020-x-sentimental-readability/readability.py
from cs50 import get_string


def main():
    text = get_string("Text: \n")

    wordcount = 0
    
    L = (100 * count_letters(text) / (count_words(text)))
    S = (100 * count_sentences(text) / (count_words(text)))
    index = 0.0588 * L - 0.296 * S - 15.8
    
    if index < 1:
        print("Before Grade 1")
    elif index > 16:
        print("Grade 16+")
    else:
        print(f"Grade {round(index)}")
      
      
def is_punctuation(char):
    if char == '?' or char == '.' or char == '!':
        return True
    else:
        return False
        

def is_letter(char):
    if 64 < ord(char) < 91 or 96 < ord(char) < 123 or char == "'" or char == '-':
        return True
    else:
        return False
        

def count_letters(text):
    letterCount = 0
    
    for char in text:
        if is_letter(char):
            letterCount += 1
        if ord(char) == 45 or ord(char) == 39:
            letterCount -= 1
        
    return letterCount
    
    
def count_words(text):
    wordCount = 0
    
    for i in range(len(text)):
        if is_letter(text[i]) and not is_letter(text[i + 1]):
            wordCount += 1
            
    return wordCount
    
    
def count_sentences(text):
    sentenceCount = 0
    
    for i in range(len(text)):
        if is_letter(text[i]) and is_punctuation(text[i + 1]):
            sentenceCount += 1
            
    return sentenceCount
    
    
main()

File: Rinnerdos-cs50-problems-2020-x-readability/readability.c
#include <stdio.h>
#include <cs50.h>
#include <string.h>
#include <ctype.h>
#include <math.h>

bool is_letter(char c);
bool is_punctuation(char c);
float count_letters(string text);
float count_words(string text);
float count_sentences(string text);


int main(void)
{
    string text = get_string("Text: ");
  
    float wordcount = 0;
    
    float L = (100 * count_letters(text) / (count_words(text)));
    float S = (100 * count_sentences(text) / (count_words(text)));
    float index = 0.0588 * L - 0.296 * S - 15.8;
    
    
    if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (index > 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %.f\n", round(index));
    }
    
}

bool is_letter(char c)
{
    if (((((int) c) > 64) && ((int) c) < 91) || ((((int) c) > 96) && ((int) c) < 123) || (int) c == 45 || (int) c == 39)
    {
        return true;
    }
    else
    {
        return false;
    }
}

bool is_punctuation(char c)
{
    if ((int) c == 33 || (int) c == 46 || (int) c == 63)
    {
        return true;
    }
    else
    {
        return false;
    }
}

float count_letters(string text)
{
    float lettercount = 0;
    for (int i = 0; i < strlen(text); i++)
    {
        if (is_letter(text[i]))
        {
            lettercount++;
        }
        if (((int) text[i] == 45 || (int) text[i] == 39))
        {
            lettercount--;
        }
    }
    return lettercount;
}

float count_words(string text)
{
    float wordcount = 0;
    for (int i = 0; i < strlen(text); i++)
    {
        if (is_letter(text[i]) && !is_letter(text[i + 1]))
        {
            wordcount++;
        }
    }
    return wordcount;
}

float count_sentences(string text)
{
    float sentencecount = 0;
    for (int i = 0; i < strlen(text); i++)
    {
        if (is_letter(text[i]) && (is_punctuation(text[i + 1])))
        {
            sentencecount++;
        }
    }
    return sentencecount;
}